File: BiWeekly Contest 51/B.cpp
class SeatManager {
public:
    priority_queue<int, vector<int>, greater<int>> heap;
    SeatManager(int n) {
        for(int i = 1; i <= n; i++){
            heap.push(i);
        }
    }
    
    int reserve() {
        int x = heap.top();
        heap.pop();
        return x;
    }
    
    void unreserve(int seatNumber) {
        heap.push(seatNumber);
    }
};

File: BiWeekly Contest 51/B.py
class SeatManager:
    def __init__(self, n: int):
        self.data = list()
        for i in range(1, n+1):
            heappush(self.data, i)

    def reserve(self) -> int:
        return heappop(self.data)

    def unreserve(self, seatNumber: int) -> None:
        heappush(self.data, seatNumber)